README.md

```diff
@@ -122,6 +122,172 @@ your container. This will allow you to login to the root account remotely. Local
+* creating more users or databases during database initialization
+
+There are basically two ways how to extend or change the image functionality:
+
+* by creating a new layered Docker image on top of this one, while using this image as base 
+* by volume mounting files during running the Docker image
+
+
+Creating a new layered Docker image on top of this Docker image
+---------------------------------------------------------------
+In order to use this Docker image as base in other layered Docker image, specify name of this one as base and then add files or run commands as you need, like this:
+```
+FROM <this-image>
+ADD mystuff /usr/bin/
+RUN /usr/bin/mystuff
+```
+
+### Example of adding a new configuration file in layered Dockerfile:
+
+To change configuration for the `mysqld` daemon and other utilities running within the container, place your new configuration files with `*.cnf` extention into `/etc/my.cnf.d` directory.
+
+Example of the layered Dockerfile:
+
+```
+FROM <this-image>
+ADD myconfig.cnf /etc/my.cnf.d/myconfig.cnf
+```
+
+Notice: Read order of configuration files in `/etc/my.cnf.d/` directory is not specified.
+
+### Adjusting database inicialization
+
+Initialization of mysql database during first start of this image is devided into several files in several directories. All directories include files that are read in alphabetical order, and thus file names are usually prefixed with a number.
+
+Concretly there are these directories within the Docker container:
+
+* `/usr/share/container-layer/mysql/pre-init/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before every inicialization of container (no matter whether the data directory is empty or not). At the time these scripts are executed, the daemon is not yet running. Example of script in this directory is a script that checks arguments specified by user.
+* `/usr/share/container-layer/mysql/post-init/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before the first start of container (i.e. in case the container is started with empty data directory). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. This directory usually includes scripts for creating new databases, users or initialize the database according to your needs.
+* `/usr/share/container-layer/mysql/pre-start/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before every start of container (no matter whether the data directory is empty or not). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. Example of script in this directory is a script that allows to change password.
```

Comment:
`pre-init` is the place where you validate variables that are used during initialization and `post-init`. You can't just shuffle with those, because we don't want the initialization to even start in case validation fails.